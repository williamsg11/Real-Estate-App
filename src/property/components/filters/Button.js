/**
 * @flow
 */
import React, {Component, PropTypes} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import colors from '../../../common/colors';

export default class Button extends Component {
  static propTypes = {
    selected: PropTypes.string.isRequired,
    range: PropTypes.array.isRequired,
    onPress: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    incrementText: PropTypes.string,
    decrementText: PropTypes.string,
    style: View.propTypes.style,
  };

  increment = () => {
    const {selected, range} = this.props;
    let arrayIndex, currentValue;
    try {
      arrayIndex = (range.indexOf(selected) + 1) % range.length;
      currentValue = range[arrayIndex];
      this.props.onPress(currentValue);
    } catch (e) {}
  };

  decrement = () => {
    const {selected, range} = this.props;
    let arrayIndex, currentValue;
    try {
      arrayIndex = range.indexOf(selected);
      arrayIndex = arrayIndex == 0 ? range.length : arrayIndex;
      currentValue = range[arrayIndex - 1];
      this.props.onPress(currentValue);
    } catch (e) {}
  };

  render() {
    const {
      title,
      titleStyle,
      selected,
      icon,
      incrementText,
      decrementText,
    } = this.props;

    return (
      <View style={styles.container}>
        <Text style={[styles.button]} onPress={() => this.decrement()}>
          {decrementText}
        </Text>
        <View style={styles.infoWrapper}>
          <View style={styles.iconWrapper}>
            <FontAwesome
              name={icon}
              color="black"
              size={20}
              style={styles.icon}
            />
            <Text style={[styles.title, titleStyle]}>{title}</Text>
          </View>
          <View style={styles.selected}>
            <Text style={styles.selectedText}>{selected}</Text>
          </View>
        </View>
        <Text style={[styles.button]} onPress={() => this.increment()}>
          {incrementText}
        </Text>
      </View>
    );
  }
}

Button.defaultProps = {
  incrementText: '+',
  decrementText: '-',
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  infoWrapper: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontWeight: '400',
    fontSize: 12,
  },
  price: {
    fontSize: 14,
  },
  icon: {
    fontSize: 20,
  },
  iconWrapper: {
    // backgroundColor:'white',
    alignItems: 'center',
  },
  selected: {
    marginLeft: 10,
  },
  selectedText: {
    color: colors.tomato,
    fontWeight: '500',
  },
  button: {
    paddingHorizontal: 10,
    fontSize: 35,
    fontWeight: '200',
    color: '#202226',
  },
});
