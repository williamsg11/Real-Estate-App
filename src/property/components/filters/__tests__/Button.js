import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from './../Button';

test('renders Button', () => {
  const tree = renderer
    .create(
      <Button
        title="Button"
        onPress={() => {}}
        selected="a"
        range={['1', '2']}
        icon="music"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
