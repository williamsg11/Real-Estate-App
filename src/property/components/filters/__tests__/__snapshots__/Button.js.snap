// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#202226",
          "fontSize": 35,
          "fontWeight": "200",
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    -
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 20,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        ÔÄÅ
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12,
              "fontWeight": "400",
            },
            undefined,
          ]
        }
      >
        Button
      </Text>
    </View>
    <View
      style={
        Object {
          "marginLeft": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#d60c3f",
            "fontWeight": "500",
          }
        }
      >
        a
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#202226",
          "fontSize": 35,
          "fontWeight": "200",
          "paddingHorizontal": 10,
        },
      ]
    }
  >
    +
  </Text>
</View>
`;
